Assets/AnkrSDK/Plugins/WalletConnectSharp.Unity/WalletConnect.cs

```diff
@@ -280,17 +357,33 @@ private void TeardownEvents()
 			_session.OnSessionDisconnect -= HandleSessionDisconnect;
 			_session.OnSessionCreated -= HandleSessionCreated;
 			_session.OnSessionResumed -= HandleSessionResumed;
+			_session.OnSend -= HandleOnSend;
+			_session.OnAccountChanged -= HandleOnAccountChanged;
+			_session.OnChainChanged -= HandleOnChainChanged;
 		}
 
-		private void HandleSessionResumed(object sender, WalletConnectSession e)
+		private void HandleOnAccountChanged(string[] accounts)
 		{
-			_resumedSessionConnected?.Invoke(e as WalletConnectUnitySession ?? _session);
+			OnAccountChanged?.Invoke(accounts);
 		}
 
-		private void HandleSessionCreated(object sender, WalletConnectSession e)
+		private void HandleOnChainChanged(int chainId)
 		{
-			_newSessionConnected?.Invoke(e as WalletConnectUnitySession ?? _session);
+			OnChainChanged?.Invoke(chainId);
+		}
 
+		private void HandleOnSend()
+		{
+			OnSend?.Invoke();
+		}
+
+		private void HandleSessionResumed()
+		{
+			
```

Comment:
missing event or state transition? If not we should remove it entirely

Reply:
I decided to keep it because we might need it later